include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_submitters.h

```diff
@@ -30,16 +33,137 @@ template <typename _KeyT, ::std::uint8_t __radix_bits, ::std::uint32_t __stage_c
+                __global_offset_data[__offset] = __presum;
+            });
+        });
+    }
+};
+
 template <bool __is_ascending, ::std::uint8_t __radix_bits, ::std::uint16_t __data_per_work_item,
           ::std::uint16_t __work_group_size, typename _KeyT, typename _KernelName>
 struct __radix_sort_onesweep_submitter;
 
+template <bool __is_ascending, ::std::uint8_t __radix_bits, ::std::uint16_t __data_per_work_item,
+          ::std::uint16_t __work_group_size, typename _KeyT, typename... _Name>
+struct __radix_sort_onesweep_submitter<__is_ascending, __radix_bits, __data_per_work_item, __work_group_size, _KeyT,
+                                       oneapi::dpl::__par_backend_hetero::__internal::__optional_kernel_name<_Name...>>
+{
+    template <typename _InKeysRng, typename _OutKeysRng, typename _GlobalHistT>
+    sycl::event
+    operator()(sycl::queue& __q, _InKeysRng& __in_keys_rng, _OutKeysRng& __out_keys_rng, _GlobalHistT* __p_global_hist,
+               _GlobalHistT* __p_group_hists, ::std::uint32_t __sweep_work_group_count, ::std::size_t __n,
+               ::std::uint32_t __stage, const sycl::event& __e) const
+    {
+        sycl::nd_range<1> __nd_range(__sweep_work_group_count * __work_group_size, __work_group_size);
+        return __q.submit([&](sycl::handler& __cgh) {
+            oneapi::dpl::__ranges::__require_access(__cgh, __in_keys_rng, __out_keys_rng);
+            auto __in_data = __in_keys_rng.data();
+            auto __out_data = __out_keys_rng.data();
+            __cgh.depends_on(__e);
+            __radix_sort_onesweep_slm_reorder_kernel<__is_ascending, __radix_bits, __data_per_work_item,
+                                                     __work_group_size, _KeyT, decltype(__in_data),
+                                                     decltype(__out_data)>
+                __sweep_kernel(__n, __stage, __in_data, __out_data, __p_global_hist, __p_group_hists);
+            __cgh.parallel_for<_Name...>(__nd_range, __sweep_kernel);
+        });
+    }
+};
+
 template <typename _KeyT, typename _KernelName>
 struct __radix_sort_copyback_submitter;
 
+template <typename _KeyT, typename... _Name>
+struct __radix_sort_copyback_submitter<_KeyT,
+                                       oneapi::dpl::__par_backend_hetero::__internal::__optional_kernel_name<_Name...>>
+{
+    template <typename _KeysTmpRng, typename _KeysRng>
+    sycl::event
+    operator()(sycl::queue& __q, _KeysTmpRng& __keys_tmp_rng, _KeysRng& __keys_rng, ::std::uint32_t __n,
+               const sycl::event& __e) const
+    {
+        return __q.submit([&](sycl::handler& __cgh) {
+            oneapi::dpl::__ranges::__require_access(__cgh, __keys_tmp_rng, __keys_rng);
+            // TODO: make sure that access is read_only for __keys_tmp_rng  and is write_only for __keys_rng
+            auto __tmp_data = __keys_tmp_rng.data();
+            auto __out_data = __keys_rng.data();
```

Comment:
You can use `__keys_tmp_rng` and `__keys_rng` directly, removing `__tmp_data` and `__out_data`. ESIMD radix sort KT relies on `data` function because of ESIMD technology limitations, which is not a case for SYCL-based kernels.